code/data_structures/src/tree/binary_tree/cartesian_tree/cartesian_tree.cpp

```diff
@@ -0,0 +1,143 @@
+/* Part of Cosmos by OpenGenus Foundation */
+
+/*
+ * c++ 11 code to construct a Cartesian tree.
+ * The method cartesianTreeSort will sort the contents of the array.
+ * InorderTraversal will return the original sequence.
+ */
+
+#include <iostream>
+#include <vector>
+#include <queue>
+
+struct Node
+{
+    int value;
+    Node *left, *right;
+    Node *parent;
+
+    Node ()
+    {
+        value = 0;
+        parent = NULL;
+        left = NULL;
+        right = NULL;
+    }
+};
+
+// Used by priority queue
+struct Compare
+{
+    bool operator() (Node *left, Node *right)
+    {
+        return left->value > right->value;
```

Comment:
You should probably check for `nullptr` to avoid invoking UB.

Reply:
I am only pushing non-null values in priority_queue so it shouldn't be a problem.